mmit/core/piecewise_function.cpp

```diff
@@ -19,161 +19,155 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "piecewise_function.h"
 
+#define TOL 1e-7
```

Comment:
I wonder if we absolutely need to introduce a numerical tolerance parameter? It would be better if we did not have to...